lib/Sema/TypeCheckType.cpp

```diff
@@ -3010,7 +3010,10 @@ Type TypeResolver::resolveCompositionType(CompositionTypeRepr *repr,
 
   for (auto tyR : repr->getTypes()) {
     Type ty = resolveType(tyR, options.withoutContext());
-    if (!ty || ty->hasError()) return ty;
+    if (!ty || ty->hasError()) {
+      repr->setResolvedType(ty);
```

Comment:
If this has some effect, can you split it off into a separate commit with a test case?

Reply:
Type representations do not store the types they represent. I suspect the chief reason is redundancy. In my case having such an ability for `CompositionTypeRepr` was important because otherwise I would have to duplicate the logic in `resolveCompositionType` to get ahold of a type that I can use to inspect existential support in `UnsupportedProtocolVisitor::visitCompositionTypeRepr`.